contracts/auctions/src/utils.rs

```diff
@@ -0,0 +1,35 @@
+use soroban_sdk::{log, Env};
+
+use crate::{
+    error::ContractError,
+    storage::{Auction, AuctionStatus},
+};
+
+pub(crate) fn check_auction_can_be_finalized(
+    env: &Env,
+    auction: &Auction,
+) -> Result<(), ContractError> {
+    if auction.status != AuctionStatus::Active {
+        log!(
+            env,
+            "Auction: Finalize auction: Cannot finalize an inactive/ended auction."
+        );
+        return Err(ContractError::AuctionNotActive);
+    }
+    if env.ledger().timestamp() < auction.end_time {
+        log!(
+            env,
+            "Auction: Finalize auction: Auction cannot be ended early"
+        );
+        return Err(ContractError::AuctionNotFinished);
+    }
+    Ok(())
+}
```

Comment:
Again, a helper used in only once place that hides actual conditions being checked.